### PR TITLE
regenerate based on latest openshift/api commit; use --trim-path-prefix

### PR DESCRIPTION
**Pull request by joelanford**

### CODE REVIEW

File: hack/update-codegen.sh [hunk 1]
```diff
@@ -63,13 +63,15 @@ for group in apiserver apps authorization build cloudnetwork image imageregistry
     "${group}:v1" \
     --go-header-file ${SCRIPT_ROOT}/hack/boilerplate.txt \
     --plural-exceptions=DNS:DNSes,DNSList:DNSList,Endpoints:Endpoints,Features:Features,FeaturesList:FeaturesList,SecurityContextConstraints:SecurityContextConstraints \
+    --trim-path-prefix github.com/openshift/client-go \
```
Comment (JoelSpeed) [inline hack/update-codegen.sh:66]: Have you tried this with `GOPATH` set and unset? Does it produce the correct result in both cases?

Comment (joelanford) [reply hack/update-codegen.sh:66]: Setting `GOPATH=~go`, it seems like generated files end up in `~/go/src/github.com/openshift/client-go`, regardless of the `--trim-path-prefix`

Unsetting GOPATH:
1.  leave `--trim-path-prefix` unset
    - generated files ended up in `./github.com/openshift/client-go`
1. set `--trim-path-prefix github.com/openshift/client-go`
    - generated files end up in `./` (correctly adding/overwriting existing files)

Perhaps we should do one of:
1. In the Makefile, set `GOPATH=./`
2. Explicitly set `--output base ./`

Comment (joelanford) [reply hack/update-codegen.sh:67]: Updated to add `--output-base ${SCRIPT_ROOT}`

Just curious: if I've got the repo cloned outside of GOPATH, is there a way right now (i.e. without `--trim-path-prefix`) to run `make generate` and have it output files to the correct places?

Ultimately that's what I'm trying to solve. So if that already works, maybe it just needs to be documented?

Comment (ncdc) [reply hack/update-codegen.sh:67]: That's why I added `--trim-path-prefix`; I don't believe it is possible without it.

Comment (JoelSpeed) [reply hack/update-codegen.sh:67]: Using `--trim-path-prefix` you can basically make the tool act on relative paths. We had this set up in openshift/api but I think I broke it, what we had working there was to get it working with `--trim-path-prefix` and then in the makefile add `GOPATH= hack/update-codegen.sh` so the whole script runs without a GOPATH set

Comment (joelanford) [reply hack/update-codegen.sh:66]: Updated to use `GOPATH=` method in the Makefile

File: Makefile [hunk 2]
```diff
@@ -22,14 +22,14 @@ examples:
 .PHONY: examples
 
 verify:
-	hack/verify-codegen.sh
+	GOPATH= hack/verify-codegen.sh
```
Comment (joelanford) [inline Makefile:25]: `verify-codegen.sh` calls `update-codegen.sh`, so this needs `GOPATH=` as well.

**Pull request merged**